src/sage/combinat/integer_vector.py

```diff
@@ -1160,10 +1323,58 @@ def rank(self, x):
         for i in range(k - 1):
             k -= 1
             n -= x[i]
-            r += binomial(k + n - 1, k)
+            r += binomial(k + n - 1, n - 1)
 
         return r
 
+    def unrank(self, x):
+        """
+        Return the element at given rank x.
+
+        INPUT:
+
+        - ``x`` -- an integer such that ``x < len(self)``
+
+        EXAMPLES::
+
+            sage: IntegerVectors(4,5).unrank(30)
+            [1, 0, 1, 0, 2]
+            sage: IntegerVectors(2,3).unrank(5)
+            [0, 0, 2]
+        """
+        if x >= len(self):
```

Comment:
Using `len(self)` seems problematic. For me
```
sage: V=IntegerVectors(100000,6); c=V.cardinality(); c; V[c-1]
```
produces
```
83345834041685416895001
...
OverflowError: cannot fit 'int' into an index-sized integer
```

How about using `cardinality` instead?